src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -46,9 +54,16 @@ public TElement[] ToArray()
 
             TElement[] array = new TElement[count];
             int[] map = SortedMap(buffer);
```

Comment:
In particular, note that both key allocation and comparison of keys happens via user-supplied code. As such those two places are places where any exception can be thrown.